src/many-kvstore/src/module/account.rs

```diff
@@ -0,0 +1,316 @@
+        self.storage.add_roles(account, args)?;
+        Ok(EmptyReturn)
+    }
+
+    fn remove_roles(
+        &mut self,
+        sender: &Address,
+        args: account::RemoveRolesArgs,
+    ) -> Result<EmptyReturn, ManyError> {
+        let account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        if !account.has_role(sender, account::Role::Owner) {
+            return Err(account::errors::user_needs_role(Role::Owner));
+        }
+        self.storage.remove_roles(account, args)?;
+        Ok(EmptyReturn)
+    }
+
+    fn info(
+        &self,
+        _sender: &Address,
+        args: account::InfoArgs,
+    ) -> Result<account::InfoReturn, ManyError> {
+        let account::Account {
+            description,
+            roles,
+            features,
+            disabled,
+        } = self
+            .storage
+            .get_account_even_disabled(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        Ok(account::InfoReturn {
+            description,
+            roles,
+            features,
+            disabled,
+        })
+    }
+
+    fn disable(
+        &mut self,
+        sender: &Address,
+        args: account::DisableArgs,
+    ) -> Result<EmptyReturn, ManyError> {
+        let account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        if !account.has_role(sender, Role::Owner) {
+            return Err(account::errors::user_needs_role(Role::Owner));
+        }
+
+        self.storage.disable_account(&args.account)?;
+        Ok(EmptyReturn)
+    }
+
+    fn add_features(
+        &mut self,
+        sender: &Address,
+        args: account::AddFeaturesArgs,
+    ) -> Result<account::AddFeaturesReturn, ManyError> {
+        if args.features.is_empty() {
+            return Err(account::errors::empty_feature());
+        }
+        let account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        account.needs_role(sender, [Role::Owner])?;
+        self.storage.add_features(account, args)?;
+        Ok(EmptyReturn)
+    }
+}
+
+impl KvStoreModuleImpl {
+    /// Verify the alternative owner is supported
+    /// Verify the sender has the rights to use this alternative owner address
+    pub(crate) fn validate_alternative_owner<R: TryInto<Role> + std::fmt::Display + Copy>(
+        &self,
+        sender: &Address,
+        alternative_owner: &Address,
+        roles: impl IntoIterator<Item = R>,
+    ) -> Result<(), ManyError> {
+        if let Some(account) = self.storage.get_account(alternative_owner) {
+            account.needs_role(sender, roles)
+        } else if alternative_owner.is_subresource() {
+            Err(error::subres_alt_unsupported())
```

Comment:
Why?

Reply:
Ah, yes.

I will re-verify, but I think we're missing some bits in `many-rs` to be able to verify that a subresource belongs to a given parent address. I didn't want to switch context and just disallowed it for now.

I can add it in a future PR.

WDYT?

Reply:
https://github.com/liftedinit/many-framework/issues/246